### PR TITLE
New filters! ⭐⭐⭐

### PR DESCRIPTION
**Pull request by homm**
This adds two new filters to `resize` method.

BOX is event faster than LINEAR because it has a smaller window (0.5) and produces very sharp images. Applicable for significant downscaling.

HAMMING costs like LINEAR, also produces very sharp images and applicable for downscaling at any range. Doesn't look good for upscaling, though.

### CODE REVIEW

**Comment by homm** [comment]
This PR is not critical for me and has relatively large change for review, so it can wait until 3.4. The most valuable part from point of view is filters comparison table in documentation. So if we decide to move this PR to 3.4 I can make another PR with the table only.

**Comment by wiredfool** [comment]
Ok, bumping to 3.4.

**Comment by homm** [comment]
Needs rebasing.

File: libImaging/Imaging.h [hunk 1]
```diff
@@ -229,9 +229,11 @@ extern void ImagingError_Clear(void);
 
 /* standard filters */
 #define IMAGING_TRANSFORM_NEAREST 0
-#define IMAGING_TRANSFORM_LANCZOS 1
+#define IMAGING_TRANSFORM_BOX 4
 #define IMAGING_TRANSFORM_BILINEAR 2
+#define IMAGING_TRANSFORM_HAMMING 5
 #define IMAGING_TRANSFORM_BICUBIC 3
+#define IMAGING_TRANSFORM_LANCZOS 1
```
Comment (wiredfool) [inline libImaging/Imaging.h]: Is there any logical order here?  It doesn't appear to be numeric, alphabetical, or order added.

Comment (homm) [inline libImaging/Imaging.h]: From less expensive and quality to more. Same as in the comparison table.

Comment (homm) [inline libImaging/Imaging.h]: @wiredfool Are you happy with this explanation?

Comment (wiredfool) [inline libImaging/Imaging.h]: Yeah, though I expect someone is going to helpfully clean up the order in a PR in a year or two.

**Comment by wiredfool** [comment]
@homm Can you write up basic release notes for this in docs/release/3.4.0.rst?

**Comment by homm** [comment]
Yes, sure. But I'd prefere to delay this on the weak before the release, to add all notes at once :)

**Comment by wiredfool** [comment]
I'd rather have it in there so that we don't miss it.

**Pull request merged**